samples/unity-of-bugs/Assets/Scripts/SentryNativeBridge.m

```diff
@@ -0,0 +1,121 @@
+void SentryNativeBridgeAddBreadcrumb(const char* timestamp, const char* message, const char* type, const char* category, int* level) {
+    NSLog(@"Native Bridge: Adding breadcrumb.");
+    
+    if (timestamp == NULL && message == NULL && type == NULL && category == NULL && level == NULL) {
+        NSLog(@"Native Bridge: Breadcrumb empty. Dropping it.");
+        return;
+    }
+    
+    [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+        SentryBreadcrumb *breadcrumb = [[SentryBreadcrumb alloc] init];
+        
+        if (timestamp != NULL) {
+            NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+            [dateFormatter setDateFormat:NSCalendarIdentifierISO8601];
+            breadcrumb.timestamp = [dateFormatter dateFromString:[NSString stringWithCString:timestamp encoding:NSUTF8StringEncoding]];
+        }
+
+        if (message != NULL) {
+            breadcrumb.message = [NSString stringWithCString:message encoding:NSUTF8StringEncoding];
+        }
+
+        if (type != NULL) {
+            breadcrumb.type = [NSString stringWithCString:type encoding:NSUTF8StringEncoding];
+        }
+        
+        if (category != NULL) {
+            breadcrumb.category = [NSString stringWithCString:category encoding:NSUTF8StringEncoding];
+        }
+        
+        if (level != NULL) {
+            breadcrumb.level = level;
+        }
+        
+        [scope addBreadcrumb:breadcrumb];
+    }];
+}
+
+void SentryNativeBridgeSetExtra(const char* key, const char* value) {
+    NSLog(@"Native Bridge: Adding extra.");
+    
+    if (key == NULL) {
+        NSLog(@"Native Bridge: Extra key empty. Dropping it.");
+        return;
+    }
+    
+    [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+        if (value != NULL) {
+            [scope setExtraValue:[NSString stringWithUTF8String:value] forKey:[NSString stringWithUTF8String:key]];
+        }
+        else {
+            [scope removeExtraForKey:[NSString stringWithUTF8String:key]];            
+        }
+    }];
+}
+
+void SentryNativeBridgeSetTag(const char* key, const char* value) {
+    NSLog(@"Native Bridge: Setting tag.");
+    
+    if (key == NULL) {
+        NSLog(@"Native Bridge: Tag key empty. Dropping it.");
+        return;
+    }
+    
+    [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+        if (value != NULL) {
+            [scope setTagValue:[NSString stringWithUTF8String:value] forKey:[NSString stringWithUTF8String:key]];
+        }
+        else {
+            [scope removeTagForKey:[NSString stringWithUTF8String:key]];
+        }
+    }];
+}
+
+void SentryNativeBridgeUnsetTag(const char* key) {
+    NSLog(@"Native Bridge: Unsetting tag.");
+    [SentrySDK configureScope:^(SentryScope * _Nonnull scope) {
+        [scope removeTagForKey:[NSString stringWithUTF8String:key]];
+    }];
+}
+
+void SentryNativeBridgeSetUser(const char* email, const char* userId, const char* ipAddress, const char* username) {
```

Comment:
`m`: Do you also have data, as in Cocoa? https://github.com/getsentry/sentry-cocoa/blob/7fb40d6e9dfb74fb444621b4475b866daabc9820/Sources/Sentry/Public/SentryUser.h#L32

Reply:
We do have an `<string, string>` dictionary for additional data that we intend to sync down at a later point.
Gets tracked via: https://github.com/getsentry/sentry-unity/issues/314